service/http.proxy/src/main/java/org/kaazing/gateway/service/http/proxy/HttpProxyService.java

```diff
@@ -38,11 +39,35 @@ public void init(ServiceContext serviceContext) throws Exception {
         if (connectURIs == null || connectURIs.isEmpty()) {
             throw new IllegalArgumentException("Missing required element: <connect>");
         }
+
+        checkForTrailingSlashes(serviceContext);
+
         HttpProxyServiceHandler handler = getHandler();
         handler.setConnectURIs(connectURIs);
         handler.initServiceConnectManager();
     }
 
+    private void checkForTrailingSlashes(ServiceContext serviceContext) {
+        Collection<URI> acceptURIs = serviceContext.getAccepts();
+        Collection<URI> connectURIs = serviceContext.getConnects();
+        Iterator<URI> acceptIterator = acceptURIs.iterator();
+        Iterator<URI> connectIterator = connectURIs.iterator();
+
+        while (acceptIterator.hasNext() && connectIterator.hasNext()) {
+            String acceptPath = acceptIterator.next().getPath();
+            String connectPath = connectIterator.next().getPath();
+            boolean acceptPathEndsInSlash = acceptPath.endsWith("/");
+            boolean connectPathEndsInSlash = connectPath.endsWith("/");
+
+            if (acceptPathEndsInSlash != connectPathEndsInSlash) {
```

Comment:
Here we have a scenario for which this is not correct. For example with a config such as:

```
    <accept>http://localhost:8018</accept>

    <connect>http://developer.kaazing.com/b/c</connect> 

```
In GatewayContextResolver getCanonicalURI will append "/" to the path in accept, and the above check will throw the exception, as the path in connect has no trailing "/". How do we now if the request adeed a "/" or not so that we can add it or not to the connect. Do we want that for the above config?